### PR TITLE
Avoid needless witgen and only generate proofs in CI.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Comment by chriseth** [comment]
I'm actually not sure if the proof generation or multiple runs of witgen is the problem.

File: pipeline/tests/pil.rs [hunk 1]
```diff
@@ -90,9 +94,7 @@ fn permutation_with_selector() {
 #[test]
 fn fibonacci() {
     let f = "pil/fibonacci.pil";
-    verify_pil(f, Default::default());
-    test_halo2(f, Default::default());
-    gen_estark_proof(f, Default::default());
```
Comment (chriseth) [inline pipeline/tests/pil.rs:95]: pil and estark would re-run witgen from scratch even though the parameters are the same. Now the pipeline is re-used for pil and estark.

File: pipeline/tests/pil.rs [hunk 2]
```diff
@@ -122,38 +124,37 @@ fn fibonacci_invalid_witness() {
...
+    test_pilcom(pipeline);
 }
 
 #[test]
 fn external_witgen_a_provided() {
     let f = "pil/external_witgen.pil";
     let external_witness = vec![("main.a".to_string(), vec![GoldilocksField::from(3); 16])];
-    run_pilcom_test_file(f, Default::default(), external_witness).unwrap();
+    let pipeline = make_prepared_pipeline(f, Default::default(), external_witness);
+    run_pilcom_test_file(pipeline).unwrap();
```
Comment (chriseth) [inline pipeline/tests/pil.rs:149]: Need to change this to a call to `test_pilcom`

**Comment by chriseth** [comment]
Proof generation time is indeed significant.

**Comment by chriseth** [comment]
@georgwiese I saw that `gen_estark_proof_with_backend_variant` calls `compute_proof()` again after setting the backend.

Is that a bug (I believe it has been there before), i.e. does the proof depend on the backend (composite / monolithic)?

**Comment by chriseth** [comment]
Maybe we should also further restrict the book tests

**Comment by leonardoalt** [comment]
> I'm actually not sure if the proof generation or multiple runs of witgen is the problem.

I'd guess both

File: pipeline/src/test_util.rs [hunk 3]
```diff
@@ -119,10 +116,18 @@ pub fn run_pilcom_with_backend_variant(
...
 
-pub fn gen_estark_proof(file_name: &str, inputs: Vec<GoldilocksField>) {
-    let pipeline = make_prepared_pipeline(file_name, inputs, Vec::new());
-    gen_estark_proof_with_backend_variant(pipeline.clone(), BackendVariant::Monolithic);
-    gen_estark_proof_with_backend_variant(pipeline, BackendVariant::Composite);
+fn should_generate_proofs() -> bool {
+    env::var("POWDR_GENERATE_PROOFS")
+        .ok()
+        .map(|v| !v.is_empty())
+        == Some(true)
```
Comment (lvella) [inline pipeline/src/test_util.rs:123]: So, `POWDR_GENERATE_PROOFS=false` or `0` is true, and `POWDR_GENERATE_PROOFS=""` is false?

I would change to `env::vars_os().any(|(k, _)| k.to_str() == Some("POWDR_GENERATE_PROOFS"))`, so if it is set to whatever, it is true, even empty or invalid UTF-8.

File: pipeline/tests/asm.rs [hunk 4]
```diff
@@ -1,22 +1,21 @@
 use powdr_backend::BackendType;
 use powdr_executor::constant_evaluator::get_uniquely_sized;
 use powdr_number::{Bn254Field, FieldElement, GoldilocksField};
+
+// TODO where was gen_estark_proof used in this file?
+
```
Comment (georgwiese) [inline pipeline/tests/asm.rs:6]:
```suggestion
```

File: pipeline/src/test_util.rs (hunk 3 above)
Comment (georgwiese) [reply pipeline/src/test_util.rs:123]: But then, can you easily un-define it? Say I have it set to some value in my environment, but I want to turn off proof generation for just one test run?

Maybe we we should only accept `true` and `false` (panic if set to any other value), and default to `false` if it is unset?

Comment (lvella) [reply pipeline/src/test_util.rs]: According to chatgpt, you can do `env -u VARIABLE_NAME command`. But it is settled now.

**Pull request merged**